### PR TITLE
Fix issue with pointer parsing

### PR DESCRIPTION
**Pull request by w32blaster**
During the development, I faced with the problem, that two properties can't be parsed, because parser fails with the message _"Type is not supported"_. Turns out, parser can't parse the pointers. As a quick solution, I manually parse two properties and assign to struct fields depending on their value. Tests are provided as well.

### CODE REVIEW

File: env/env.go [hunk 1]
```diff
@@ -1,25 +1,40 @@
 package env
 
 import (
-	"github.com/EVODelavega/goq/aws"
+	"os"
+
+	goqaws "github.com/EVODelavega/goq/aws"
```
Comment (EVODelavega) [inline env/env.go]: aws specific imports in a general "env" package? That's not good. Just create a config package that is AWS specific with a `Get` func.

File: env/env.go [hunk 2]
```diff
@@ -1,25 +1,40 @@
...
 	}
 
-	config := &aws.Config{}
+	config := &goqaws.Config{}
 	if err := env.Parse(config); err != nil {
 		return nil, err
 	}
 
+	// Parse manually properties, that are presented as pointers
+	config.QueueUrl = getStringPtr(os.Getenv("AWS_QUEUE_URL"))
```
Comment (EVODelavega) [inline env/env.go]: the env package has customer parser support. Also,  the queue url or topic ARN might not be set, in which case `os.LookupEnv` might be the better function to use

**Review by EVODelavega — CHANGES_REQUESTED** [review]

File: env/env.go (hunk 2 above)
Comment (w32blaster) [reply env/env.go]: Good point

File: aws/sqs.go [hunk 3]
```diff
@@ -284,7 +286,7 @@ func (p *pubSQS) sendMessage(ctx context.Context, msg goq.BaseMsg, delay *int64)
 		MessageBody:       &body,
 		MessageAttributes: msgAttr,
 	}
-	if out, err := p.client.SendMessageWithContext(ctx, &in, nil); err != nil {
+	if out, err := p.client.SendMessageWithContext(ctx, &in); err != nil {
```
Comment (w32blaster) [inline aws/sqs.go:287]: Damn, I wanted to make it as separate PR :disappointed: 
This `nil` causes nil pointer error at this line:
`github.com/aws/aws-sdk-go/service/sqs/api.go:1573` So instead of nil, I pass empty vararg

File: aws/sqs.go [hunk 4]
```diff
@@ -174,7 +174,9 @@ func (s *subSQS) consumeLoop(ctx context.Context) {
 			if err != nil {
 				log.Errorf("Failed to receive messages: %+v", err)
 			} else {
-				log.Infof("Received %d messages", len(rec.Messages))
+				if len(rec.Messages) > 0 {
```
Comment (EVODelavega) [inline aws/sqs.go]: Logging the fact that no messages are received is something I'd still do. When testing, at least you can see when the consumer is working, regardless of there being messages to process or not

File: env/env.go [hunk 5]
```diff
@@ -15,11 +19,26 @@ func Get() (*aws.Config, error) {
...
 	}
 
 	config.Credentials = credentials
 
 	return config, nil
 }
+
+// Optional properties are presented as pointers allowing to have nil's when value were not provided.
+// This simple function returns pointer only in case when string value is not empty
+func fnGetStringPointer(v string) (interface{}, error) {
```
Comment (EVODelavega) [inline env/env.go]: `fn`-prefix looks rather pointless
